### DISCUSSION TITLE
Trouble reading observation file

### DISCUSSION THREAD

**Discussion by lswenson** [dtcenter/METplus#2468] [File I/O] [answered]
I'm trying to use MODE on precipitation data from the GFS and StageIV. I've regridded both datasets to the same 0.5 degree grid and saved them to netcdf files. I've tried to include all the necessary metadata in the netcdf files but I keep running into the error message "Trouble reading observation file." It's unclear to me what this means or how to fix it. Any help would be greatly appreciated. I've shared the relevant netcdf files and my config file to the ftp server. Thank you very much!

**Comment by CPKalb** [comment, marked as answer]
Hi:

Both your forecast and observation files are not actually in a format that MET/MODE can read.  MET accepts netCDF files that are CF-compliant, output from WRF pinterp, or in MET's own netCDF format.  The files you have are closest to CF-compliant.  However, MET does not try to read them as a CF compliant file because they lack the global attribute "Conventions" as described in the [requirements](https://met.readthedocs.io/en/latest/Users_Guide/data_io.html#requirements-for-cf-compliant-netcdf).  MET's default is to read files in MET's own netCDF format which in this case won't work, as your data is not in that format.

You can force MET/MODE to try to read them as a CF-compliant netCDF file by setting the following in your configuration file:
MODE_FCST_FILE_TYPE = NETCDF_NCCF
MODE_OBS_FILE_TYPE = NETCDF_NCCF

When I do this, MODE reads the data, but I get the following warnings and information:
WARNING: NcCfFile::open() -> could not determine the valid time, using 0.
WARNING: 
WARNING: 
WARNING: NcCfFile::open() -> could not determine the valid time, using 0.
WARNING: 
DEBUG 1: Forecast File: 20230105_fcast2.nc
DEBUG 1: Observation File: 20230106_obs2.nc
DEBUG 2: 
DEBUG 2: NcCfFile::getData(NcVar *, const LongArray &, DataPlane &) const -> data was flipped to north.
DEBUG 2: 
DEBUG 2: NcCfFile::getData(NcVar *, const LongArray &, DataPlane &) const -> data was flipped to north.

In this case, you don't have a time variable in either file as described in the above requirements.  So, MODE will set all output to use the default time (19790101_000000).  If you are planning on running more than one time, the output files from these different times will all be labeled with the default time and will overwrite each other.  So, a time would be needed in the files to prevent this from happening.

Additionally, it looks like MODE is flipping your grid.  I'm not certain whether this is desirable.  I ran a test using plot_data_plane to see how MET is reading the data and verify if it is correct.  The commands to do that are as follows:
plot_data_plane 20230105_fcast2.nc testfcst.ps name="gfs"; level="(*,*)"; file_type=NETCDF_NCCF;
plot_data_plane 20230106_obs2.nc testobs.ps 'name="verification"; level=(*,*); file_type = NETCDF_NCCF;'

Here is the output:
Forecast
![testfcst](https://github.com/dtcenter/METplus/assets/22180338/3efe889c-89a4-446d-b210-dc9626c647c9)

Obs:
![testobs](https://github.com/dtcenter/METplus/assets/22180338/72057b37-2d5f-4122-b1ec-f959c446f20e)

Is this what the data should look like?


Christina

**Reply by lswenson** [reply]
Thank you so much, this was very helpful! Adding the file_type flags produces the results I was expecting. That is what my data looks like so I think the grid flipping is desirable.

**Reply by CPKalb** [reply]
Great.  I'm going to go ahead and lock the discussion to prevent future posts.  If you have additional questions feel free to submit a new discussion!

Christina
